test/action_view/component_test.rb

```diff
@@ -17,6 +17,33 @@ def test_render_inline_with_old_helper
     assert_html_matches "<div>hello,world!</div>", result.css("div").to_html
   end
 
+  def test_zero_arity_initializers
+      result = render_inline(ZeroArityComponent)
+
+      assert_html_matches "<div>hello,world!</div>", result.to_html
+  end
+
+  def test_zero_arity_initializers_with_hash_syntax
+    result = render_inline(component: ZeroArityComponent, locals: {})
+
+    assert_html_matches "<div>hello,world!</div>", result.to_html
+  end
+
+
+  def test_zero_arity_initializers_with_hash_syntax_and_no_locals
+    result = render_inline(component: ZeroArityComponent)
+
+    assert_html_matches "<div>hello,world!</div>", result.to_html
+  end
+
+  def test_zero_arity_initializer_arguments_not_dropped
+    exception = assert_raises ArgumentError do
+      render_inline(ZeroArityComponent, unnecessary_attr: "Hello")
+    end
+
+    assert_includes exception.message, "wrong number of arguments (given 1, expected 0)"
+  end
```

Comment:
Can we add a test for rendering a zero arity component via the `to_component_class` interface?

Reply:
No problem.

I think it should always raise an error (since the object is always passed through to the component)

Reply:
Here's the test:

https://github.com/github/actionview-component/pull/171/files#diff-1ead8b7b852d11443af608850a661ed2R39-R45